### ISSUE TITLE
[Issue]: Image filename pattern defaults to include whole prompt in image name

### ISSUE THREAD

**Issue by DrMeo** [vladmandic/automatic#1543]
### Issue Description

Behavior now:
When settings>Image Options>"image filename patterns" is left blank the image filename includes a large chunk of the prompt, including spaces. 

This can lead to images having too long of filenames in Windows if the full install %PATH% including the outputs folders and filenames is too long. Such instances are harder to work with and move around since Explorer can't handle them gracefully.

Expected behavior:
Be like how it is in older commits where settings>Image Options>"image filename patterns" is left blank the image name is only numbers and dashes ( if "add numbers to filenames is also selected) with no spaces, and no prompt included in the file name. 

I also noticed that the "[" and "]" characters are not escaped when used as an option alone, "[]" will almost mimic the older commit versions of filenaming with no prompt inclusion, but will add in "[]" in place of the second number string to the filename.

For example: 

Older commit before the naming changed - 00001-2990041446.png
New naming style with the patterns set to "[]" so avoid the prompt being int he file name -  00001-[].png

### Version Platform Description

Windows 11 Version	10.0.22621 Build 22621
Google Chrome
21:56:53-960145 INFO     Python 3.10.8 on Windows
21:56:53-976153 INFO     Version: 0d3092c2 Fri Jun 30 17:22:42 2023 -0400
21:56:54-305482 INFO     Latest published version: 634c98e3dcb8b695888d436aa2bf4bb78a7986a2 2023-07-01T00:49:16Z
21:56:54-305482 INFO     nVidia CUDA toolkit detected
21:56:55-277843 INFO     Torch 2.0.0+cu118
21:56:55-293473 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
21:56:55-293473 INFO     Torch detected GPU: NVIDIA GeForce RTX 3070 Ti VRAM 8192 Arch (8, 6) Cores 48

### Acknowledgements

- [X] I have read the above and searched for existing issues

**Comment by brknsoul** [comment]
There are already a couple solutions for this. In SD.Next, there's a slider at the bottom of Settings > Image Options to adjust how many words are included in [prompt].

The other solution is to enable Windows (10/11) Long File Name support, which increases the length limit of path+name from 260 characters to 32,768 characters;
https://www.thewindowsclub.com/how-to-enable-or-disable-win32-long-paths-in-windows-11-10

**Comment by DrMeo** [comment]
Default behavior shouldn't be changed, even if there is a slider. Especially since there is no indication that the defaults have changed in the program. Up until the change - not sure when exactly it happened, I just noticed the change in my test environment, otherwise I use a version that is over a month old for my normal day to day usage, if you left the filename field blank you got a certain behavior.  Good UI design practices dictate either 1: don't change defaults, users can do that themselves if needed, or 2: indicate the new defaults and how to set the behavior back to the old behavior. 

The slider should only have to be used if you explicitly use the [promptwords] setting ( or other settings ) in the image filename patterns box.

**Comment by vladmandic** [comment]
> Default behavior shouldn't be changed

Sorry, but that is NOT realistic, especially since this is not even semver versioned, you're using live master branch!
And even if it were semver version, I would be bumping minor version pretty much daily as things are moving too fast to have patch-only releases.

>  Good UI design practices dictate either 1: don't change defaults, users can do that themselves if needed, or 2: indicate the new defaults and how to set the behavior back to the old behavior.

That is not true in general except for software that doesn't evolve - and if that happens, I will announce that I'm giving up. 

And lets not get into flame wars. You had a request and that is fine - the rest is unnecessary noise!

Anyhow, I've added few extra fields to filename template: `[seq]`, `[image_hash]' and `[uuid]`
If you want just image sequence as filename, set template to be just `[seq]`.
And I've changed default template from `[seed]-[prompt_spaces]` to `[seq]-[prompt_words]`
